### ISSUE TITLE
Linux framebuffer support

### ISSUE THREAD

**Issue by crocket** [alacritty/alacritty#2757]
Would it be possible to support linux framebuffer?

RetroArch can be run on linux framebuffer.
GZDoom also runs on linux framebuffer.

**Comment by chrisduerr** [comment]
Why would you want to see this supported? What usecase would this have for you?

**Comment by crocket** [comment]
* Seeing file names in various languages on linux framebuffer.
* GPU-accelerated console on linux framebuffer

**Comment by chrisduerr** [comment]
I'll be honest, usually I'd just say that this is out of scope for the project and there's little reason to spend significant amount of effort for such a niche usecase. However there is some plan to explore implementing the kmsmod backend in our upstream windowing library.

If this support is added upstream, it would likely be trivial to support this in Alacritty. So I'd say this depends solely on our upstream dependency. If they decide it's not worth it, I wouldn't pursue it myself.

So for everyone interested, I recommend subscribing to the upstream issue:
https://github.com/rust-windowing/winit/issues/1006

I'll keep this one around as a tracking issue from the Alacritty perspective.

**Comment by nixpulvis** [comment]
I'll just leave this here as some potential motivation: https://kmandla.wordpress.com/2010/05/28/fbterm-birth-of-the-cool-for-the-console/

I'm actually wondering how well this would work to allow my framebuffer to support modern fonts.

**Comment by laanwj** [comment]
> I suspect mesa's gbm can make this work.

Right, linux framebuffer through `/dev/fb0` is deprecated. The modern way to render directly to framebuffer is through KMS + gbm. It's fairly straightforward to set up OpenGL rendering that way (see [kmscube](https://gitlab.freedesktop.org/mesa/kmscube/)].

This was attempted with [Kmscon](https://en.wikipedia.org/wiki/Kmscon) but this isn't maintained anymore.

The most difficult part will actually be input handling, I think. Using evdev input directly is acceptable for games, but for something with complex multi-lingual text input requirements it's highly non-trivial.

**Comment by thealio** [comment]
> * Seeing file names in various languages on linux framebuffer. Proper unicode rendering support.
> 
>     * GPU-accelerated console on linux framebuffer
> 
>     * Color theme on linux framebuffer
>       
>       * Default linux virtual terminal colors are painful to my eyes.
> 
> 
> I suspect mesa's gbm can make this work.

+ 1 for framebuffer support  for the same reasons
This implementation could really open interesting scenarios for Alacritty
Maybe can you  take some code for fbterm for implement this feature?

**Comment by kchibisov** [comment]
@thealio we already had alacritty in frame buffer, but I don't remember exactly where the branch was. It's just no one has time to work on that.

**Comment by thealio** [comment]
> @thealio we already had alacritty in frame buffer, but I don't remember exactly where the branch was. It's just no one has time to work on that.

Thank you forma you answer.
So, is it currently possible to run alacritty in the framebuffer?And if yes, how can I run it?
If not, I'd like to see the current code for framebuffer implemented into current alacritty branch

**Comment by kchibisov** [comment]
alacritty doesn't need anything to support frame buffer other than `cfg` a bunch of x11/wayland stuff. The main code should be here https://github.com/rust-windowing/winit/pull/2795

**Comment by thealio** [comment]
> alacritty doesn't need anything to support frame buffer other than `cfg` a bunch of x11/wayland stuff. The main code should be here [rust-windowing/winit#2795](https://github.com/rust-windowing/winit/pull/2795)

Thank you for your answer
So, is it possible to use alacritty in the tty as a framebuffer terminal (like fbterm for example)? 
What packages I currently need to use alacrity in the tty? 
And how I can start it from the tty?

Yesterday I tried by typing alacrity from the tty, and it gave errors

I'm sorry if I ask this, but it's still not clear to me

**Comment by kchibisov** [comment]
it's not possible now, it could be possible if all related patches will be merged.

You could also, theoretically, go and try to build it yourself, but I won't guide you here wrt kms stuff, because it's not stable and will waste my time...

it's not a framebuffer though, it's kms in your case, it'll be a bit different, but faster.

**Comment by thealio** [comment]
> it's not possible now, it could be possible if all related patches will be merged.
> 
> You could also, theoretically, go and try to build it yourself, but I won't guide you here wrt kms stuff, because it's not stable and will waste my time...
> 
> it's not a framebuffer though, it's kms in your case, it'll be a bit different, but faster.

Is this merge planned as experimental feature anytime soon?
This is a great feature to add in my opinion

**Comment by oech3** [comment]
KMSCON is actively maintained now. So I think we can close this issue (if Alacritty cannot be faster than KMSCON on console).

**Comment by thealio** [comment]
I think kmscon has not all the goodies of alacritty (colors, great vim mode etc)
I think a fanebuffer support would still be great

2026-09-19T03:01:29Z oech3 ***@***.***>:

**Comment by oech3** [comment]
How about Alacritty on [cage](https://www.hjdskes.nl/projects/cage/)?

**Comment by crocket** [comment]
I think expecting maintainers to support linux frame buffer is a step too far. Thus, closing it.
